src/main.rs

```diff
@@ -1,17 +1,20 @@
 #![feature(type_alias_impl_trait)]
+#![feature(generic_associated_types)]
```

Comment:
Oh, this is not necessary btw, make sure you're on the latest version of cargo (`rustup update nightly`)